core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1919,17 +1952,14 @@ private static String toSql(RelNode root, SqlDialect dialect) {
         + "on \"t1\".\"product_id\" = \"t3\".\"product_id\" or "
         + "(\"t1\".\"product_id\" is not null or "
         + "\"t3\".\"product_id\" is not null)";
-    // Some of the "IS NULL" and "IS NOT NULL" are reduced to TRUE or FALSE,
+    // Some of the "IS NULL" and "IS NOT NULL" are reduced,
     // but not all.
```

Comment:
Is this change expected?

Reply:
Yes, it's by intention -- with this change [1], rexNodes which are alwaysTrue or alwaysFalse is actually reduced out, rather than leave a TRUE or FALSE in the condition.

[1] https://github.com/apache/calcite/pull/1647/files#diff-79ccdc597ccf355912fea1a60683eea2R155